extensions/vertx-http/runtime/src/main/java/io/quarkus/devui/runtime/jsonrpc/json/JsonMapper.java

```diff
@@ -52,7 +52,11 @@ public <T> T fromString(String json, Class<T> target) {
 
                     @Override
                     public <T> T fromValue(Object json, Class<T> target) {
-                        return target.cast(fromValue.apply(json, target));
+                        if (target.isPrimitive()) {
+                            return (T) json;
```

Comment:
Does that work everytime? Why wouldn't the `else` branch for primitive?

Reply:
This works fine, if not there , JsonPRC methods can not use primitives for input parameters.